### ISSUE TITLE
Voice input consistently crashes the game for many Linux users

### ISSUE THREAD

**Issue by chloecormier** [tf2classic/tf2classic-issue-tracker#11]
This is pretty inconsistent and hard to pin down, but we've had a lot of reports that, across distributions, whether you're using PulseAudio or PipeWire, pressing the voice chat bind crashes the game.

I have a hunch this only affects newer distros, since it's only confirmed on Ubuntu 22.10, Debian Sid, and Fedora 35 so far. For me, it happens inconsistently, and tends to crash the rest of the Steam client with it (likely because it's using Steam Voice). It also doesn't always crash for me, and if it works once then it works for the rest of the playsession. Other reporters say it crashes consistently, no matter what. These might be separate issues?

This issue isn't fixed by using the Steam Linux Runtime. It _is_ fixed by running the Windows version of the game through Proton. Affected users say it doesn't affect live TF2.

**Comment by Gwenily** [comment]
Seconding this issue. I'd love to troubleshoot but i'm honestly not even sure where to begin. No crash information is given in `console.log` for me.
Pop!_OS 22.04

**Comment by AZMCode** [comment]
I can consistently trigger this bug, (my steam client doesn't crash) and can provide a single line of error log and a minidump. Would that be useful?

**Comment by AZMCode** [comment]
In my case the error always happens after the following line is logged to console:

```
src/tcmalloc.cc:278] Attempt to free invalid pointer 0x296e01a0
```

**Comment by AZMCode** [comment]
Are there any command line options/debugging tools I could use to provide better info?
I basically have only told you it's a double-free.

**Comment by AZMCode** [comment]
Can confirm switching to pipewire does the game no favors

**Comment by AZMCode** [comment]
From @Teownik on Discord, reposting here for ease of access
```
>I searched on GitHub and some guy said that setting value in game console sv_voicecodec to steam and launching local server fixes the problem
>And I can confirm that. Changing value and launching local server and then use microphone on Linux fixes that problem
>But the problem is that most of the servers uses in sv_voicecodec value vaudio_celt and I think the main problem is a codec that crashes the game

**Comment by AZMCode** [comment]
Confirming myself in a moment

**Comment by AZMCode** [comment]
Can confirm it works.
Only if you change the cvar before starting the server though. Would be nice if you all could temporarily change the codec in the default servers while this issue is resolved.

**Comment by AZMCode** [comment]
In contact with VF4 for the time being to see if this can be implemented as a stopgap measure in the main servers

**Comment by AZMCode** [comment]
Some have reported disabling microphone gain in settings works, but it doesn't in my case

**Comment by sapphonie** [comment]
Fixed in 2.0.4 with defaulting steam codec. Also an issue in live.
